src/lang/KclSinglton.tsx

```diff
@@ -233,13 +262,17 @@ class KclManager {
     this.ast = ast
     if (code) this.code = code
   }
-  setCode(code: string) {
+  setCode(code: string, shouldWriteFile = true) {
+    if (shouldWriteFile) {
+      // use the normal code setter
+      this.code = code
+      return
+    }
     this._code = code
     this._codeCallBack(code)
-    localStorage.setItem(PERSIST_CODE_TOKEN, code)
   }
-  setCodeAndExecute(code: string) {
-    this.setCode(code)
+  setCodeAndExecute(code: string, shouldWriteFile = true) {
```

Comment:
Generally, when the code is updated we want to save the file as well. Except when the file is first loaded because the user opened a project.